docs/types/value-types.rst

```diff
@@ -463,14 +463,23 @@ There is no additional semantic meaning added to a number literal containing und
 the underscores are ignored.
 
 Number literal expressions retain arbitrary precision until they are converted to a non-literal type (i.e. by
-using them together with a non-literal expression or by explicit conversion).
+using them together with anything else than a number literal expression (like boolean literals) or by explicit conversion).
 This means that computations do not overflow and divisions do not truncate
 in number literal expressions.
 
 For example, ``(2**800 + 1) - 2**800`` results in the constant ``1`` (of type ``uint8``)
 although intermediate results would not even fit the machine word size. Furthermore, ``.5 * 8`` results
 in the integer ``4`` (although non-integers were used in between).
 
+.. warning::
+    There are cases, such as those involving the ternary operator, where implicit conversions to sub-types of a smaller size can happen.
+    (For example, a rational number 100 will get converted to uint8 because uint8 is the smallest sized type that is sufficient to accommodate it)
+    This can lead to unintended overflows during calculations further down the pipeline.
+
+    For example ``255 + (true ? 1 : 0)`` will revert due to arithmetic overflow.
+    The reason is that ``(true ? 1 : 0)`` is of ``uint8`` type, which forces the addition to be performed in ``uint8`` as well,
+    and 256 exceeds the range allowed for this type.
```

Comment:
Here's my take:
```suggestion
    While most operators produce a literal expression when applied to literals, there are certain operators that do not follow this rule:
    - Ternary operator (``... ? ... : ...``),
    - Array subscript (``<array>[<index>]``).

    You might expect expressions like ``255 + (true ? 1 : 0)`` or ``255 + [1, 2, 3][0]`` to be equivalent to using the literal 256 directly, but in fact they are computed within type ``uint8`` and due to this can overflow.
```